### PR TITLE
docs: add terragrunt example

### PR DESCRIPTION
**Pull request by hugorut**
Adds support for terragrunt pipeline example. Currently installs terragrunt with bash as terragrunt task looks relatively unused.

### CODE REVIEW

File: examples/terragrunt/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    steps:
+      - task: TerraformInstaller@0
+        displayName: Install Terraform
+
+      - bash: |
+          INSTALL_LOCATION="$(Build.SourcesDirectory)/bin"
+          mkdir -p ${INSTALL_LOCATION}
+          
+          LATEST_URL=$(curl -sL  https://api.github.com/repos/gruntwork-io/terragrunt/releases  | jq -r '.[0].assets[].browser_download_url' | egrep 'linux.*amd64' | tail -1)
```
Comment (aliscott) [inline examples/terragrunt/README.md:27]: Should we use this URL `https://github.com/gruntwork-io/terragrunt/releases/latest/download/terragrunt_linux_amd64`? The GitHub API has rate limits for anonymous requests so might fail without an GitHub token in the header at times.

File: examples/terragrunt/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          apiKey: $(API_KEY)
+
+      - bash: infracost breakdown --path=examples/terragrunt/code --format=json --out-file=/tmp/infracost.json
+        displayName: Run Infracost
+        
+      - task: InfracostComment@0
+        displayName: Post the comment
+        inputs:
+          path: /tmp/infracost.json
+          behavior: update # Create a single comment and update it. See https://github.com/infracost/actions/tree/master/comment for other options
```
Comment (aliscott) [inline examples/terragrunt/README.md:47]: Should we point to https://github.com/infracost/infracost-azure-devops#comment-options (assuming this link will work in master in the future)?

**Review by aliscott — APPROVED** [review]
LGTM

File: examples/terragrunt/README.md (hunk 2 above)
Comment (hugorut) [reply examples/terragrunt/README.md:47]: good shout

File: examples/terragrunt/README.md (hunk 1 above)
Comment (hugorut) [reply examples/terragrunt/README.md:27]: yup - I'll update

Comment (hugorut) [reply examples/terragrunt/README.md]: resolved

File: examples/terragrunt/README.md (hunk 2 above)
Comment (hugorut) [reply examples/terragrunt/README.md]: resolved

**Pull request merged**